go/src/koding/klient/machine/machinegroup/mount.go

```diff
@@ -402,3 +399,62 @@ func (g *Group) Umount(req *UmountRequest) (res *UmountResponse, err error) {
 		Mount:   m,
 	}, nil
 }
+
+// InspectMountRequest defines machine group mount inspect request.
+type InspectMountRequest struct {
+	// Identifier is a string that identifiers requested mount. It can be either
+	// mount ID or local path which is going to be inspected.
+	Identifier string `json:"identifier"`
+
+	// Sync indicates whether inspect should attach sync history or not.
+	Sync bool `json:"sync"`
+}
+
+// InspectMountResponse defines machine group mount inspect response.
+type InspectMountResponse struct {
+	// History contains information about recently synchronized files.
+	History []*history.Record `json:"history"`
+}
+
+// InspectMount gets detailed information about mount current state.
+func (g *Group) InspectMount(req *InspectMountRequest) (*InspectMountResponse, error) {
+	if req == nil {
+		return nil, errors.New("invalid nil request")
+	}
+
+	// Get mount ID from identifier.
+	mountID, err := g.getMountID(req.Identifier)
+	if err != nil {
+		return nil, err
+	}
+
+	res := &InspectMountResponse{}
+	if !req.Sync {
+		return res, nil
+	}
+
+	if sc, err := g.sync.Sync(mountID); err == nil {
+		if res.History, err = sc.History(); err != nil {
+			g.log.Error("Cannot get mount %s syncing history: %s", mountID, err)
```

Comment:
because there are plans to have other options than `Sync` and I want this function to be greedy and get as much data as it can.